### PR TITLE
either: implement Bifunctor

### LINKED ISSUE #98: Mapping the left side of an Either
Let's explore if these functions in [`data.either`](https://github.com/folktale/data.either) would be useful in Sanctuary:
- [`orElse`](https://github.com/folktale/data.either/blob/master/lib/either.js#L313-L315)
- [`bimap`](https://github.com/folktale/data.either/blob/master/lib/either.js#L380-L394)
- [`leftMap`](https://github.com/folktale/data.either/blob/master/lib/either.js#L397-L408)

### PR DESCRIPTION
**Pull request by davidchambers**
Closes #98

You're likely the best person to review this, @scott-christopher. :)

### CODE REVIEW

**Comment by scott-christopher** [comment]
This LGTM. :mushroom:

File: index.js [hunk 1]
```diff
@@ -1543,6 +1543,35 @@
...
+  //.
+  //. Takes two functions and returns:
+  //.
+  //.   - a Left whose value is the result of applying the first function
+  //.     to this Left's value if `this` is a Left; otherwise
+  //.
+  //.   - a Right whose value is the result of applying the second function
+  //.     to this Right's value.
+  //.
+  //. Similar to [`Either#map`](#Either.prototype.map), but supports mapping
```
Comment (rjmk) [inline index.js]: Reciprocal link?

Comment (davidchambers) [inline index.js]: :zap:

**Comment by davidchambers** [comment]
This has changed as a result of #305 being merged. I'd appreciate a review of the laws. :)

File: test/internal/laws/Bifunctor.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+var bimap = require('./bimap');
+var compose = require('./compose');
+var forall = require('./forall');
+var id = require('./id');
+
+
+module.exports = function(equals) {
```
Comment (safareli) [inline test/internal/laws/Bifunctor.js]: ✅

File: test/Either/Either.js [hunk 3]
```diff
@@ -63,6 +63,24 @@ suite('Either', function() {
 
   });
 
+  suite('Bifunctor laws', function() {
```
Comment (safareli) [inline test/Either/Either.js]: ✅

**Pull request merged**